source/_integrations/zwave_js.markdown

```diff
@@ -20,56 +20,66 @@ ha_codeowners:
+This is the recommended approach if you're running the default version of Home Assistant, including the supervisor and you do not require any advanced configuration/options but just want your Z-Wave devices available to control within Home Assistant.
 
-## Configuration
+### 2. The Z-Wave JS to MQTT add-on installed available from the community add-on store
+
+This is the recommended approach if you're running the default version of Home Assistant, including the supervisor and you like to have maximum control over your Z-Wave network and it's nodes. Despite what the name suggests, you can actually run this addon without MQTT enabled. In that case it will provide you with a full fledged, attractive and feature complete UI to manage your Z-Wave nodes and settings. The Z-Wave JS server is **included** in Zwave JS to MQTT, you enable it on the Z-Wave tab within the UI. If you install the add-on from the community repo, this will be done for you automatically.
+
+### 3. The Z-Wave JS to MQTT docker container
+
+This is the recommended approach if you're running the Home Assistant Core yourself, without the supervisor and it's convenient add-ons. See the [zwavejs2m1tt documentation](https://zwave-js.github.io/zwavejs2mqtt/#/getting-started/quick-start) for instructions. Important note: make sure you run a recent version of the docker image, as that contains the requires WS server option.
+
+### 4. Run the Z-Wave server yourself
+
+This is considered a very advanced scenario where you run the (nodejs) Z-Wave JS Server (or zwavejs2mqtt) yourself. Installation and maintaining this is out of scope for this documentation. See the [Z-Wave JS server](https://github.com/zwave-js/zwave-js-server) or [zwavejs2mqtt](https://github.com/zwave-js/zwavejs2mqtt) Github repo's for information.
 
-This integration can be configured using the integrations in the
-Home Assistant frontend.
+### Important note
 
-Menu: **Configuration** -> **Integrations**.
+When you start the Z-Wave JS server, it will begin interviewing your entire Z-Wave network, depending on the size of your network (how many nodes) this can take a while, especially when started for the very first time. Information about your devices is stored in cache files by Z-Wave JS. Be aware that (re)starting the Z-Wave server will cause your network to be (partially) unresponsive until the interview process is done. While your Z-Wave mesh is permanently stored on your stick, the additional metadata is not, so when you loose the cache files (for example by switching between any of the above mentioned ways to run the server) all your nodes will have to be re-interviewed again before they can be properly controlled. You can speed up this process by manually waking up your battery powered devices. Most of the times this is a simple press on the button on those devices (see manual). In any way it is not needed to exclude/re-include devices from the mesh. Just be patient with this.
```

Comment:
```suggestion
When you start the Z-Wave JS server, it will begin interviewing your entire Z-Wave network, depending on the size of your network (how many nodes) this can take a while, especially when started for the very first time. Information about your devices is stored in cache files by Z-Wave JS. Be aware that (re)starting the Z-Wave server will cause your network to be (partially) unresponsive until the interview process is done.

While your Z-Wave mesh is permanently stored on your stick, the additional metadata is not, so when you lose the cache files (for example by switching between any of the above-mentioned ways to run the server) all your nodes will have to be re-interviewed again before they can be properly controlled. You can speed up this process by manually waking up your battery-powered devices. Most of the time this is a press on the button on those devices (see manual). In any way, it is not needed to exclude/re-include devices from the mesh. Just be patient with this.
```